src/main/java/seedu/task/logic/parser/AddParser.java

```diff
@@ -20,12 +22,17 @@
 
 public class AddParser implements Parser {
 
+    public static final Prefix descriptionPrefix = new Prefix("/desc");
+    public static final Prefix deadlinePrefix = new Prefix("/by");
+    public static final Prefix durationPrefix = new Prefix("/from");
+    
     public AddParser() {}
      
     private static final Pattern TASK_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter prefixes
             Pattern.compile("(?<name>[^/]+)"
-                    + "(?: /desc (?<description>[^/]+))*"
+                    + "(?<description>(?: /desc [^/]+)*)"
                     + "(?<deadline>(?: /by [^/]+)*)"
+                    + "(?<description2>(?: /desc [^/]+)*)"
```

Comment:
because the order is now flexible, the regex needs to detect if the description is after deadline. I think i'll look into a better way to check for presence of task or event